### PR TITLE
Introduce IR v11

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Introduce IR v11
     - Only v11 IRs contain rt-info section
 - Restore the right order of parameter/result operations
 - TI and other subgraph operation are serialized with v10 order due to the ticket: 67158

### Ticket:
 - 67157

### CODE REVIEW

File: inference-engine/src/transformations/include/transformations/serialize.hpp [hunk 1]
```diff
@@ -32,17 +31,25 @@ class TRANSFORMATIONS_API StreamSerialize;
...
-    Serialize(std::ostream & xmlFile, std::ostream & binFile,
-              Version version = Version::IR_V10,
-              std::map<std::string, ngraph::OpSet> custom_opsets = {});
+    Serialize(std::ostream& xmlFile, std::ostream& binFile,
+              std::map<std::string, ngraph::OpSet> custom_opsets,
+              Version version = Version::UNSPECIFIED);
+    Serialize(std::ostream& xmlFile, std::ostream& binFile,
+              Version version = Version::UNSPECIFIED);
 
     Serialize(const std::string& xmlPath, const std::string& binPath,
```
Comment (ilya-lavrenov) [inline inference-engine/src/transformations/include/transformations/serialize.hpp:48]: will we deprecated ctor with custom_opsets in new API?
Looks like end customer have to use new RTTI and don't use custom opsets for serialize

File: inference-engine/src/transformations/src/transformations/serialize.cpp [hunk 2]
```diff
@@ -1024,25 +1040,33 @@ bool pass::Serialize::run_on_function(std::shared_ptr<ngraph::Function> f) {
...
-            break;
+        auto version = static_cast<int64_t>(m_version);
+        auto& rt_info = f->get_rt_info();
+        if (rt_info.count("version")) {
+            auto version_var = std::dynamic_pointer_cast<VariantWrapper<int64_t>>(rt_info.at("version"));
+            version = version_var->get();
+        }
+
+        if (version != static_cast<int64_t>(m_version) && m_version != Serialize::Version::UNSPECIFIED)
+            throw ngraph_error("Cannot serialize function to incompatible IR version");
```
Comment (ilya-lavrenov) [inline inference-engine/src/transformations/src/transformations/serialize.cpp:1051]: can we extend the error like `you are trying to serialize IR v10 as IR 11` or smth like this?

File: inference-engine/src/transformations/src/transformations/serialize.cpp [hunk 3]
```diff
@@ -1124,6 +1159,19 @@ bool ngraph::pass::StreamSerialize::run_on_function(std::shared_ptr<ngraph::Func
...
     };
+    auto version = static_cast<int64_t>(m_version);
+    auto& rt_info = f->get_rt_info();
+    if (rt_info.count("version")) {
+        auto version_var = std::dynamic_pointer_cast<VariantWrapper<int64_t>>(rt_info.at("version"));
+        version = version_var->get();
+    }
+
+    if (version != static_cast<int64_t>(m_version) && m_version != Serialize::Version::UNSPECIFIED)
+        throw ngraph_error("Cannot serialize function to incompatible IR version");
```
Comment (ilya-lavrenov) [inline inference-engine/src/transformations/src/transformations/serialize.cpp:1170]: the same

File: inference-engine/tests/functional/inference_engine/ir_serialization/rt_info_serialization.cpp [hunk 4]
```diff
@@ -75,3 +75,138 @@ TEST_F(RTInfoSerializationTest, all_attributes) {
...
+    }
+
+    pass::Manager m;
+    m.register_pass<pass::Serialize>(m_out_xml_path, m_out_bin_path, pass::Serialize::Version::IR_V11);
+    m.run_passes(function);
+
+    auto core = InferenceEngine::Core();
+    auto net = core.ReadNetwork(m_out_xml_path, m_out_bin_path);
+    auto f = net.getFunction();
+
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/inference_engine/ir_serialization/rt_info_serialization.cpp:197]: should we check that IR v11 is read as IR v10 in old API? or we need to do it only for cases when `old_api_map` is applied?

File: docs/template_plugin/src/template_executable_network.cpp [hunk 5]
```diff
@@ -203,7 +203,7 @@ void TemplatePlugin::ExecutableNetwork::Export(std::ostream& modelStream) {
     // Note: custom ngraph extensions are not supported
     std::map<std::string, ngraph::OpSet> custom_opsets;
     std::stringstream xmlFile, binFile;
-    ngraph::pass::Serialize serializer(xmlFile, binFile, ngraph::pass::Serialize::Version::IR_V10, custom_opsets);
+    ngraph::pass::Serialize serializer(xmlFile, binFile, custom_opsets, ngraph::pass::Serialize::Version::IR_V11);
```
Comment (ilya-lavrenov) [inline docs/template_plugin/src/template_executable_network.cpp:206]: I believe we should serialize as is - we information which is inside `ov::Function`?
What if IR v10 is passed?

File: inference-engine/src/inference_engine/src/cnn_network_ngraph_impl.cpp [hunk 6]
```diff
@@ -489,8 +489,8 @@ StatusCode CNNNetworkNGraphImpl::serialize(const std::string& xmlPath,
         ngraph::pass::Manager manager;
         manager.register_pass<ngraph::pass::Serialize>(xmlPath,
                                                        binPath,
-                                                       ngraph::pass::Serialize::Version::IR_V10,
-                                                       custom_opsets);
+                                                       custom_opsets,
+                                                       ngraph::pass::Serialize::Version::IR_V10);
```
Comment (ilya-lavrenov) [inline inference-engine/src/inference_engine/src/cnn_network_ngraph_impl.cpp:493]: also - can we use information from the function itself?

Comment (ilya-lavrenov) [reply inference-engine/src/inference_engine/src/cnn_network_ngraph_impl.cpp:493]: Discussed offline that if we read IR v11 in old API - it's always downgraded to IR v10

File: ngraph/core/include/openvino/core/function.hpp [hunk 7]
```diff
@@ -263,6 +264,12 @@ class OPENVINO_API Function : public std::enable_shared_from_this<Function> {
 
     /// \brief Return a variable by specified variable_id.
     ngraph::VariablePtr get_variable_by_id(const std::string& variable_id) const;
+    RTMap& get_rt_info() {
+        return m_rt_info;
+    }
+    const RTMap& get_rt_info() const {
+        return m_rt_info;
+    }
```
Comment (ilya-lavrenov) [inline ngraph/core/include/openvino/core/function.hpp:272]: can we add a test that this information is copied during `clone_function`?

File: inference-engine/src/transformations/include/transformations/serialize.hpp (hunk 1 above)
Comment (ilyachur) [reply inference-engine/src/transformations/include/transformations/serialize.hpp:47]: I propose to do it in the next PR with public transformation

File: docs/template_plugin/src/template_executable_network.cpp (hunk 5 above)
Comment (ilya-lavrenov) [reply docs/template_plugin/src/template_executable_network.cpp:206]: can we remove explicit `IR_V11` and serializer will always use latest version?

**Comment by ilyachur** [comment]
@ilya-lavrenov @GlebKazantaev @jane-intel @itikhono please take a look

**Review by jane-intel — APPROVED** [review]
🔥

**Review by itikhono — APPROVED** [review]

**Pull request merged**